src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,35 +74,44 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                Type declaringType = testMethodDeclaringType;
+                Type declaringType;
 
-                // We have qualified type name with the assembly name, something like
-                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
-                // We don't use '.' as separator in such case and use '!' because the qualified type name can have '.' to include the type namespace.
-                if (conditionMemberName.IndexOf('!') > 0)
+                if (calleeType != null)
+                {
+                    declaringType = calleeType;
+                }
+                else
                 {
-                    // get the method name
-                    string[] symbols = conditionMemberName.Split('!');
-                    if (symbols.Length == 2)
+                    declaringType = testMethodDeclaringType;
+
+                    // We have qualified type name with the assembly name, something like
+                    // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
+                    // We don't use '.' as separator in such case and use '!' because the qualified type name can have '.' to include the type namespace.
+                    if (conditionMemberName.IndexOf('!') > 0)
                     {
-                        conditionMemberName = symbols[1];
-                        Type requestedType = Type.GetType(symbols[0]);
-                        if (requestedType != null)
+                        // get the method name
+                        string[] symbols = conditionMemberName.Split('!');
```

Comment:
So we are still going to support the qualified name scenario? I imagine this is to cover the case where we could have multiple conditions that live under different assemblies? For example one that is under PlatformDetection and other that is under the test class itself?